src/main/java/fr/openent/competences/service/impl/DefaultNoteService.java

```diff
@@ -2516,39 +2545,49 @@ private void setJOResultWithList(JsonObject resultJO, List<NoteDevoir> notesList
-            @Override
-            public void handle(Either<String, SortedMap<String, JsonObject>> event) {
-                if (!event.isRight()) {
-                    log.error(event.left().getValue());
-                    handler.handle(new Either.Left(event.left()));
-                } else {
-                    SortedMap<String, JsonObject> mapRespMatTeacher = event.right().getValue();
-                    JsonArray matieresResult = new fr.wseduc.webutils.collections.JsonArray();
-                    for (Map.Entry<String, Set<String>> mapEntry : mapAllidMatAndidTeachers.entrySet()) {
-                        String idMatAllMats = mapEntry.getKey();
-                        JsonObject matiereJson = (JsonObject) mapRespMatTeacher.get(idMatAllMats);
-                        if(mapIdMatListMoyByEleve.containsKey(idMatAllMats)) {
-                            JsonObject statClass = utilsService.calculMoyenneParDiviseur(
-                                    mapIdMatListMoyByEleve.get(idMatAllMats),true);
-                            matiereJson.put("moyClass", statClass.getDouble("moyenne"));
-                            matiereJson.put("moyMinClass", statClass.getDouble("noteMin"));
-                            matiereJson.put("moyMaxClass", statClass.getDouble("noteMax"));
-                        } else{
-                            matiereJson.put("moyClass", "");
-                            matiereJson.put("moyMinClass", "");
-                            matiereJson.put("moyMaxClass", "");
+        if( mapAllidMatAndidTeachers.isEmpty() && mapIdMatListMoyByEleve.isEmpty()){
+            JsonObject subjectsResult = new JsonObject();
+            JsonArray subjects = new JsonArray();
+
+            subjectsResult.put("matieres", subjects);
+            subjectsResult.put ("nbDeMatieres", subjects.size());
+            handler.handle(new Either.Right<>(subjectsResult));
+
+        } else {
+            utilsService.getLibelleMatAndTeacher(mapAllidMatAndidTeachers, new Handler<Either<String, SortedMap<String, JsonObject>>>() {
+                @Override
+                public void handle (Either<String, SortedMap<String, JsonObject>> event) {
+                    if (!event.isRight()) {
+                        log.error(event.left().getValue());
+                        handler.handle(new Either.Left(event.left()));
+                    } else {
+                        SortedMap<String, JsonObject> mapRespMatTeacher = event.right().getValue();
+                        JsonArray matieresResult = new fr.wseduc.webutils.collections.JsonArray();
+                        for (Map.Entry<String, Set<String>> mapEntry : mapAllidMatAndidTeachers.entrySet()) {
+                            String idMatAllMats = mapEntry.getKey();
+                            JsonObject matiereJson = (JsonObject) mapRespMatTeacher.get(idMatAllMats);
```

Comment:
C'est déjà un JsonObject selon ta déclaration : ton cast est redondant tu peux le supprimer

```suggestion
                            JsonObject matiereJson = mapRespMatTeacher.get(idMatAllMats);
```